### ISSUE TITLE
Bug: locations sent twice

### ISSUE THREAD

**Issue by jsaedtler** [OpenLocate/openlocate-ios#71]
After installing the latest git version of OpenLocate, I receive almost all data twice. The locations have the exact same timestamp and location, it happens for passive, regular, visit_entry & visit_exit events.
It didn't happen while testing in the simulator. iPhone Models tested: 6s, 6s plus, 7s plus, iOS versions tested: 11.1, 11.2 all betas.

**Comment by kurabi** [comment]
Did this occur for just a few locations or does it constantly send all data?

This may occur if the server fails to return a HTTP 200 acknowledging receipt of data. So the client will try to send the data again (with any new additional data).

**Comment by jsaedtler** [comment]
I installed the SDK in a swift app I created from scratch, no other features or functionality included. Most of the events arrive twice, only few of them once. I don't see any events arriving more than twice.

I use the AWS API gateway to consume the events, and it returns a HTTP 200: 

```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 110
Connection: keep-alive
Date: Tue, 28 Nov 2017 09:10:21 GMT
x-amzn-RequestId: f5e1a225-d41b-11e7-8180-b1387b707581
X-Amzn-Trace-Id: Root=1-5a1d27fd-8bbe09ce8cda76888467da9f
X-Cache: Miss from cloudfront
Via: 1.1 c4ada86230c95b165d889d1f1d10389d.cloudfront.net (CloudFront)
X-Amz-Cf-Id: 6Y0Zyr4n-YJrCAzbIXn4xq3v7gTI8mO6iy__RyTKHGwbCOmM7MgwDg==
```

It looks a bit like only the latest batch of locations doesn't get removed from the send queue, like this:

```
1. sdk collects locations in batch 1
2. sdk sends batch 1
3. sdk continues to collect locations, this time in batch 2
4. sdk sends batch 1 & 2, and removes batch 1
5. sdk continues to collect locations, this time in batch 3
6. sdk sends batch 2 & 3, and removes batch 2
...
```

**Comment by kurabi** [comment]
This has been fixed as part of PR #76.
